scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ForkedJvmTaskExecutionCommandCreator.java

```
@@ -0,0 +1,124 @@
+import org.ow2.proactive.scheduler.task.context.TaskContext;
+import org.ow2.proactive.scheduler.task.context.TaskContextVariableExtractor;
+import org.ow2.proactive.scheduler.task.executors.forked.env.command.JavaPrefixCommandExtractor;
+import org.ow2.proactive.scripting.ForkEnvironmentScriptResult;
+import org.ow2.proactive.scripting.ScriptResult;
+import com.google.common.base.Strings;
+
+public class ForkedJvmTaskExecutionCommandCreator implements Serializable {
+    private final static String javaHomePostfixJavaExecutable = File.separatorChar + "bin" + File.separatorChar + "java";
+    private final TaskContextVariableExtractor taskContextVariableExtractor = new TaskContextVariableExtractor();
+    private final JavaPrefixCommandExtractor javaPrefixCommandExtractor = new JavaPrefixCommandExtractor();
+
+    /**
+     * Creates a command to start a task inside a java virtual machine.
+     *
+     * @param taskContext                   TaskContext object describing the task.
+     * @param forkEnvironmentScriptResult   Result from a running fork environment script. If it is
+     *                                      of instance {@link ForkEnvironmentScriptResult}, the script return
+     *                                      variables will be used for the construction of the command.
+     * @param serializedContextAbsolutePath The serialized TaskContext object which will be read by the virtual
+     *                                      machine to run the task.
+     * @return A List, empty if the TaskContext is null, otherwise filled with a command.
+     * @throws Exception If the {@link TaskContextVariableExtractor} could not extract all variables from the
+     *                   TaskContext.
+     */
+    public List<String> createForkedJvmTaskExecutionCommand(TaskContext taskContext,
+            ScriptResult forkEnvironmentScriptResult, String serializedContextAbsolutePath) throws Exception {
```

Do we need to throw an exception here? is it something recoverable?if not, better throw a runtimeexception in the body and leave the method signature cleaner